packages/rich-text/src/create.js

```diff
@@ -104,6 +105,26 @@ function toFormat( { tagName, attributes } ) {
 	};
 }
 
+const RichTextInternalData = Symbol( 'RichTextInternalData' );
```

Comment:
what's the point of this? are we trying to hide the internal data and prevent people from accessing it?

Reply:
I'm not sure yet. I want to make sure that the formats array especially is not too exposed [because I want to get rid of it](https://github.com/WordPress/gutenberg/issues/29870). Additionally, it would be nice if this class exposed all or most of the rich text manipulation functions (slice, split, insert, remove, etc.). Not sure if I want to do it in this PR or not, I want to make sure they are the right APIs.

Reply:
In any case, even if we rewrite the `formats` array, we'll probably have to maintain a getter for back compat...